packages/tools/devtools/devtools-core/src/data-visualization/VisualTree.ts

```diff
@@ -68,6 +68,21 @@ export interface VisualNodeBase {
 	 * Consumers of this value should attempt to handle unrecognized values gracefully.
 	 */
 	nodeKind: VisualNodeKind | string;
+
+	/**
+	 * The "kind" of tree node.
+	 *
+	 * @remarks
+	 *
+	 * This will generally be {@link VisualNodeKind}, but for forwards/backwards compatability reasons,
+	 * we allow this to be an arbitrary string.
+	 * Consumers of this value should attempt to handle unrecognized values gracefully.
+	 */
+
+	/**
+	 * (optional) The types of edits allowed
+	 */
+	editProps?: { editTypes?: EditType[] };
```

Comment:
I would recommend extracting the type info here into an interface - that way we have a place to document the properties and potentially expand on them in the future. I generally recommend against inline typing like this except for really trivial things.

Reply:
I'm not exactly sure what you are suggesting. Would this look like "HasFluidObjectId"?

Reply:
Oh, sorry. I mean extracting `{ editTypes: EditType[] }` into an interface. That way the code here can be `editProps?: EditProps;` and we can document the type and its property.